### PR TITLE
Fix plot composition

### PR DESCRIPTION
**Pull request by akhenry**
Closes https://github.com/nasa/openmct/issues/6178

### Describe your changes:
1. The composition provider [was tracking composition itself](https://github.com/nasa/openmct/pull/6206/files#diff-bb4c368d5df65905cfd33d9f8197ab096456a8c8fcd2b4ff043c7dd2b3d4fa95L103), in addition to the object's composition model. This is a recipe for staleness. It does this so that it can [calculate a diff between the old composition and the new composition](https://github.com/nasa/openmct/pull/6206/files#diff-2bb282eaf202a92b703140413ddf24eae5905f8795aebdee14dde154db268e48L235). I have implemented an overdue feature to the object API to pass the old value into mutation observer callbacks as an additional argument, alongside the new value which was already being passed in. This way we don't need to track composition separately.
2. The Create Action [was not using mutables](https://github.com/nasa/openmct/pull/6206/files#diff-5600ef72928273fb4cb379f66bb7c60d0c241ae3cd9ed3ef14b97e207b445ddeL62). I don't know to what extent (if any) this was contributing to our problems, but it might be a source of subtle bugs, which I have now fixed.
3. The Create Button was [re-fetching the navigated object](https://github.com/nasa/openmct/pull/6206/files#diff-c92ac78cbb068522ccbb63188bf89ecc3d98306c89aaa2878519292b15a816a0L82), which is redundant.
4. The `DefaultCompositionProvider` was also [directly updating the composition property on the object model _before_ calling the mutation API](https://github.com/nasa/openmct/pull/6206/files#diff-bb4c368d5df65905cfd33d9f8197ab096456a8c8fcd2b4ff043c7dd2b3d4fa95L175). This makes it impossible to track changes, so I have removed this. It now works on a copy of the object's composition.
5. The Stacked Plot was using [an invalid key in a v-for](https://github.com/nasa/openmct/pull/6206/files#diff-58e7f8ed29e685b3b82fc5627b13e0574f85bc2639b4e0e77f62ad6f3662fc47L39) which was sometimes resulting in plots being duplicated after composition changes. This has been fixed.
6. Fixed existing tests. The tests for the composition API had so much mocking that they weren't really testing anything at all.
7. Added an E2E test to catch regression of this particular bug in stacked plots.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/ui/layout/CreateButton.vue [hunk 1]
```diff
@@ -74,23 +73,9 @@ export default {
             this.openmct.menus.showSuperMenu(x, y, this.sortedItems, menuOptions);
         },
         create(key) {
-            // Hack for support.  TODO: rewrite create action.
-            // 1. Get contextual object from navigation
-            // 2. Get legacy type from legacy api
-            // 3. Instantiate create action with type, parent, context
-            // 4. perform action.
-            return this.openmct.objects.get(this.openmct.router.path[0].identifier)
```
Comment (shefalijoshi) [inline src/ui/layout/CreateButton.vue:82]: We didn't lose the context by removing this get right?

File: src/plugins/plot/stackedPlot/StackedPlot.vue [hunk 2]
```diff
@@ -35,10 +35,10 @@
     />
     <div class="l-view-section">
         <stacked-plot-item
-            v-for="object in compositionObjects"
-            :key="object.id"
+            v-for="objectWrapper in compositionObjects"
+            :key="objectWrapper.keyString"
```
Comment (shefalijoshi) [inline src/plugins/plot/stackedPlot/StackedPlot.vue:39]: Thanks!

File: src/ui/layout/CreateButton.vue (hunk 1 above)
Comment (akhenry) [reply src/ui/layout/CreateButton.vue:82]: I might be misunderstanding the issue here, but this get looks redundant to me. 

We were `get` the currently navigated object, which would always just return the domain object itself, without context.

We already have the navigated object, so there was no need to `get` it again. Maybe there's some nuance I'm missing here?

**Comment by davetsay** [comment]
@shefalijoshi re-reviewing now

**Review by shefalijoshi — APPROVED** [review]
This looks fine. There is some more work needed to properly fix stacked plots, but in general the composition fixes are great!
Thanks @akhenry

**Pull request merged**